### PR TITLE
Adopt new designs for the authentication pages (log in, password reset, etc)

### PR DESCRIPTION
**Pull request by PaarthAgarwal**
I've started adopting new Figma designs for sign-in page.
https://github.com/wagtail/wagtail/issues/2309

This PR has a long way to go, but I think it'll be a step in a good direction.
Link to figma file:https://www.figma.com/file/h67EsVXdWsfu38WGGxWfpi/Wagtail-Design-System?node-id=6899%3A41010

Before:
Screenshot from 2022-07-30 19-26-35

Right now:

Screenshot from 2022-08-06 03-32-19
Screenshot from 2022-08-06 03-32-37
Screenshot from 2022-08-06 03-32-51
Screenshot from 2022-08-06 03-33-01
Screenshot from 2022-08-06 03-33-11

Goal:
Screenshot from 2022-07-30 19-27-23

### CODE REVIEW

**Comment by PaarthAgarwal** [comment]
Made some progress today. Getting closer

File: client/scss/layouts/_login.scss [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
-
 .login {
-  margin-inline-start: 0;
-  height: 100%;
+  background: theme('colors.primary.DEFAULT');
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  padding: 0px 390px 170px;
```
Comment (lb-) [inline client/scss/layouts/_login.scss:10]: Try to use the theme spacing values here. Makes it feel more consistent and the spacing values are in REMs which is better for users that may have zoomed in.

File: client/scss/layouts/_login.scss [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
+  gap: 10px;
```
Comment (lb-) [inline client/scss/layouts/_login.scss:11]: Use spacing theme here. But also check compatibility with safari on iPhone

https://caniuse.com/?search=gap

File: client/scss/layouts/_login.scss [hunk 3]
```diff
@@ -1,18 +1,28 @@
...
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    gap: 30px;
+    isolation: isolate;
+    background: none;
+    border: none;
+    flex: none;
+    order: 6;
```
Comment (lb-) [inline client/scss/layouts/_login.scss:24]: It's not good practice to use order to re-order content if it can be avoided. It creates a problem for accessibility where the focus order does not match the visual order. There are exceptions to this guide but it's best to get the DOM in the order we want first.

File: client/scss/layouts/_login.scss [hunk 4]
```diff
@@ -23,32 +33,60 @@ $desktop-nice-padding: 100px;
...
 
   h1 {
-    font-weight: 300;
-    font-size: 2em;
-    line-height: 1em;
-    color: $color-white;
+    color: theme('colors.primary.DEFAULT');
     white-space: nowrap;
+    font-weight: 800;
+    font-size: 30px;
```
Comment (lb-) [inline client/scss/layouts/_login.scss:39]: Use theme values for font weight and font sizes

File: client/scss/layouts/_login.scss [hunk 5]
```diff
@@ -23,32 +33,60 @@ $desktop-nice-padding: 100px;
...
+    line-height: 130%;
```
Comment (lb-) [inline client/scss/layouts/_login.scss:40]: Avoid line height if you can. It creates some complexity in the styles. 
Also maybe this would suit using `@apply` - for the h1 styles. See other usage for example

File: client/scss/layouts/_login.scss [hunk 6]
```diff
@@ -23,32 +33,60 @@ $desktop-nice-padding: 100px;
...
+    text-align: center;
+    display: flex;
+    align-items: center;
+    flex: none;
+    flex-grow: 0;
+    z-index: 1;
```
Comment (lb-) [inline client/scss/layouts/_login.scss:46]: Why is this needed? This creates a new stacking context and can cause styles to be overly complex.

File: client/scss/layouts/_login.scss [hunk 7]
```diff
@@ -23,32 +33,60 @@ $desktop-nice-padding: 100px;
...
+    margin: 0px;
   }
```
Comment (lb-) [inline client/scss/layouts/_login.scss:48]: When doing margin none. Use `0` (no units, not px)

File: client/scss/layouts/_login.scss [hunk 8]
```diff
@@ -93,36 +141,11 @@ $desktop-nice-padding: 100px;
...
-    }
-
-    &__link {
-      color: $color-white;
-
-      @include media-breakpoint-up(sm) {
-        color: $color-link;
-      }
-    }
+    padding-left: 4.3em;
```
Comment (lb-) [inline client/scss/layouts/_login.scss:145]: Avoid padding left. Instead use padding start or just padding. This needs to work for RTL users also.

File: wagtail/admin/templates/wagtailadmin/login.html [hunk 9]
```diff
@@ -59,12 +59,18 @@ <h1>{% block branding_login %}{% trans "Sign in to Wagtail" %}{% endblock %}</h1
...
                             <li class="checkbox">
-                                <div class="field">
-                                    <label><input name="remember" type="checkbox" />{% trans "Remember me" %}</label>
-                                </div>
+                                <span>
+                                    <label><input name="remember" type="checkbox" />{% trans "Remember me" %} {% if show_password_reset %}
+                                        <span class="help">
+                                            <a class="help__link" href="{% url 'wagtailadmin_password_reset' %}" aria-label="{% trans 'Reset your password' %}">{% trans "Forgot your password?" %}</a>
+                                        </span>
+                                    {% endif %}</label>
```
Comment (lb-) [inline wagtail/admin/templates/wagtailadmin/login.html:68]: Try to ensure your Django template nesting is neat. It's hard to read this change - where did the if start?

File: client/scss/layouts/_login.scss [hunk 10]
```diff
@@ -23,32 +33,60 @@ $desktop-nice-padding: 100px;
...
-    ul {
-      @include unlist();
-    }
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    padding: 50px 60px 40px;
+    gap: 30px;
+    isolation: isolate;
```
Comment (lb-) [inline client/scss/layouts/_login.scss:57]: Be sure you need this. It's probably good to try to get your design working without any z-index / isolate where possible. Will make life simpler

File: client/scss/layouts/_login.scss [hunk 11]
```diff
@@ -23,32 +33,60 @@ $desktop-nice-padding: 100px;
...
-    font-size: 1.5em;
-    padding: 1.1em 2.4em;
-    height: 3.5em;
+    display: flex;
+    flex-direction: row;
+    justify-content: center;
+    align-items: center;
+    padding: 12px 18px;
+    gap: 10px;
+    width: 340px;
```
Comment (lb-) [inline client/scss/layouts/_login.scss:81]: Be careful declaring fixed widths and even fixed heights - it may come back to cause issues. 

Specially the width. 

Think about your layout as going outside in or inside out. And width 100% filling up the gaps. 

This field will likely overflow on a small device with < 400px width.

**Review by lb- — COMMENTED** [review]
General feedback
- use theme variables wherever possible
- maybe add some more useful classes, a lot of the styles are duplicated (flex etc) scss can help you by setting different classes for the same styles. Or you can add some more practical classes
- remember that the figma may be fixed but browsers are not, this needs to work well on devices with 350px viewports all the way to over 2000px - be sure to check the responsiveness of this design. 


Great start!

**Comment by lb-** [comment]
@PaarthAgarwal can you please change this to a draft. 

As noted yesterday - this code will likely change somewhat with Thibaud's form styling rework.

File: client/scss/layouts/_login.scss [hunk 12]
```diff
@@ -23,32 +21,43 @@ $desktop-nice-padding: 100px;
...
 
   h1 {
-    font-weight: 300;
-    font-size: 2em;
-    line-height: 1em;
-    color: $color-white;
+    color: theme('colors.primary.DEFAULT');
     white-space: nowrap;
+    font-weight: 800;
+    font-size: 30px;
```
Comment (lb-) [inline client/scss/layouts/_login.scss:27]:
```suggestion
    font-size: theme('fontSize.30');
```

File: client/scss/layouts/_login.scss [hunk 13]
```diff
@@ -23,32 +21,43 @@ $desktop-nice-padding: 100px;
   }
 
   h1 {
-    font-weight: 300;
-    font-size: 2em;
-    line-height: 1em;
-    color: $color-white;
+    color: theme('colors.primary.DEFAULT');
     white-space: nowrap;
+    font-weight: 800;
```
Comment (lb-) [inline client/scss/layouts/_login.scss:26]:
```suggestion
    font-weight: theme('fontWeight.extraBold');
```

File: client/scss/layouts/_login.scss [hunk 14]
```diff
@@ -23,32 +21,43 @@ $desktop-nice-padding: 100px;
   }
 
   h1 {
-    font-weight: 300;
-    font-size: 2em;
-    line-height: 1em;
-    color: $color-white;
+    color: theme('colors.primary.DEFAULT');
     white-space: nowrap;
```
Comment (lb-) [inline client/scss/layouts/_login.scss:25]: I know this style was here but I recommend against this - it might cause overflow issues for longer translations.

**Comment by lb-** [comment]
@PaarthAgarwal  as per discussion today - amazing work on this so far - here are some extra notes & feedback

# Account views

## Useful Links

- https://a11y-101.com/development/skip-link
- https://www.w3.org/WAI/tutorials/forms/notifications/#using-the-main-heading
- https://www.w3.org/WAI/tutorials/page-structure/regions/

## Issues found

- Skip link button styling issue
- Focus outline on main is not showing correctly, but it looks like there is maybe a `.login .wrapper` padding left/right that is not needed
- forgot password link should be on its own line, despite the design (make it a p not a span)
- consider devices a bit better, maybe less padding on < sm device by a bit
- .wrapper height: 100vh should be removed for the account pages (may need to override the base)
- logo - could use the existing approach (see the 404 page) - but need to document this


## New issues to raise (LB

- [ ] DOM structure of these pages
- [ ] DOM structure of the password_required etc (out of scope)
- [ ] Way too much JS loaded for just the log in page

## Files

- wagtail/admin/views/account.py (contains all the views)
- LoginView - wagtail/admin/templates/wagtailadmin/login.html
- LogoutView - not really a view on its own, but is a URL that shows a message on the log in page `localhost:8000/admin/logout/`
- PasswordResetView - wagtailadmin/account/password_reset/form.html - http://localhost:8000/admin/password_reset/
- PasswordResetDoneView - "wagtailadmin/account/password_reset/done.html" - http://localhost:8000/admin/password_reset/done/ (shows after you have submitted the reset password form)
- PasswordResetConfirmView - "wagtailadmin/account/password_reset/confirm.html" - http://localhost:8000/admin/password_reset/confirm/NA/b9khxb-5ca661e00d93e139a3c701f6ffc48710/ (must be sent to the user, need to manually create) ALSO - http://localhost:8000/admin/password_reset/confirm/NA/set-password/ will show an invalid state of this form with a message
- PasswordResetCompleteView - "wagtailadmin/account/password_reset/complete.html" - http://localhost:8000/admin/password_reset/complete/ (shows when you complete the password reset step)
- REMOVE THIS FILE wagtail/admin/templates/wagtailadmin/account/change_password.html

## Code Snippet - background

We could declare the background twice to ensure that we fall back to the plain purple if not supported but in my testing on Safari 15 it works great.

```css
  --w-login-background-gradient: radial-gradient(
    100% 300.28% at 100% 0%,
    theme('colors.primary.DEFAULT') 0%,
    #2e1f5e 32.94%,
    #2e1f5e 49.9%,
    #2e1f5e 50.2%,
    #007d7e 100%
  );
  /* add backup background for older versions of Safari (<15) */
  background: theme('colors.primary.DEFAULT');
  background: var(
    --w-login-background-gradient,
    theme('colors.primary.DEFAULT')
  );
```

## Code Snippet - logo

We could use the existing convention taken from the 404 page (just the bird but that is probably better than having the name in the image that cannot be translated).

```
{% block branding_logo %}
    <div class="MAKE THIS BETTER">
        <img src="{% versioned_static 'wagtailadmin/images/wagtail-logo.svg' %}" alt=""/>
    </div>
{% endblock %}
```

**Comment by lb-** [comment]
@PaarthAgarwal here is a proposed DOM structure rework that will be better for accessibility, more flexible and can be consistently adopted across the other log in / auth pages.

* body (as currently done in the PR - contains the background colour)
  * skip link
  * wrapper div (exists already on core templates) - is the white / rounded corners container
    * header
      * h1 (title)
      * `block above_login` (put this under the h1, it is more appropriate to have h1 first
      * page level messages in a div with aria-assertive=true (note: use absolute positioning to put at the top of the page maybe) - such as success 
    * main (contains the main `form`)
    * footer (contains `block below_login` and then the logo block)

We can discuss more - or move this DOM structure change to a new issue - but have a read of the accessibility links above and see what you think.

**Comment by PaarthAgarwal** [comment]
I fixed the skiplink width issue but with the new gradient I'm unsure if it offers sufficient contrast
Screenshot from 2022-08-05 15-40-26

**Comment by lb-** [comment]
The skip link should be ok. 
We have the focus outline offers a clear separation and the text on the button colour should have high enough contrast.

File: wagtail/admin/templates/wagtailadmin/login.html [hunk 15]
```diff
@@ -64,15 +64,16 @@ <h1>{% block branding_login %}{% trans "Sign in to Wagtail" %}{% endblock %}</h1
...
-                                                <a class="help__link" href="{% url 'wagtailadmin_password_reset' %}" aria-label="{% trans 'Reset your password' %}">{% trans "Forgotten password?" %}</a>
-                                            </span>
-                                        {% endif %}
                                     </label>
                                 </div>
                             </li>
-
+                            {% if show_password_reset %}
+                                <li class="help">
+                                    <div>
```
Comment (zerolab) [inline wagtail/admin/templates/wagtailadmin/login.html:72]: drive-by comment: can we avoid `div`-itis like this unless strictly necessary. and if necessary, accompany it with a `{# contextual comment as to why #}` ?

Comment (PaarthAgarwal) [reply wagtail/admin/templates/wagtailadmin/login.html]: I added it initially to keep the dom structure consistent with the above content. Now, with Thibaud's PR, it has been refactored.

File: wagtail/admin/templates/wagtailadmin/account/password_reset/confirm.html [hunk 16]
```diff
@@ -22,15 +22,17 @@ <h1>{% trans "Set your new password" %}</h1>
...
-            <div class="messages">
-                <ul>
-                    <li class="error">
-                        {% trans "The password reset link was invalid, possibly because it has already been used." %}
-                    </li>
-                </ul>
-            </div>
-            <a href="{% url 'wagtailadmin_password_reset' %}" class="button">{% trans 'Request a new password reset' %}</a>
+            <form class="content" method="post" novalidate>
+                <h1>{% trans "Invalid password reset link" %}</h1>
```
Comment (lb-) [inline wagtail/admin/templates/wagtailadmin/account/password_reset/confirm.html:26]: I think this form is not needed.  Maybe an outer wrapper div.content for the while thing would be better?

File: wagtail/admin/templates/wagtailadmin/login.html [hunk 17]
```diff
@@ -31,29 +31,24 @@ <h1>{% block branding_login %}{% trans "Sign in to Wagtail" %}{% endblock %}</h1
...
                 {% endblock %}
-
-                <div class="w-mt-4 w-mb-4">
-                    {% block submit_buttons %}
-                        <button type="submit" class="button button-longrunning" data-clicked-text="{% trans 'Signing in…' %}">{% icon name="spinner" %}<em>{% trans 'Sign in' %}</em></button>
-                    {% endblock %}
-                </div>
-
-                {% if show_password_reset %}
-                    <a class="reset-password" href="{% url 'wagtailadmin_password_reset' %}" aria-label="{% trans 'Reset your password' %}">{% trans "Forgotten password?" %}</a>
```
Comment (lb-) [inline wagtail/admin/templates/wagtailadmin/login.html:55]: best to keep the original translation - even though the design had a different label - I think this can be left as is for the sake of translations and it's not really a design decision.

**Review by lb- — COMMENTED** [review]
hey - I have taken a few liberties on this and applied the comments as defined.

This was close but just needed a bit more polish and a simpler DOM structure approach.

File: client/scss/layouts/_login.scss [hunk 18]
```diff
@@ -2,152 +2,124 @@
...
+  margin-top: 0;
+  margin-bottom: 1.1em;
+  padding: 10px 0;
+}
 
-    ul {
-      margin: 0;
+.messages {
+  margin: 1em 0;
+  z-index: 5;
```
[the remark this answers is not in the document]
Comment (PaarthAgarwal) [reply client/scss/layouts/_login.scss:79]: I don't remember adding it🤔

[the remark this answers is not in the document]
Comment (PaarthAgarwal) [reply client/scss/layouts/_login.scss:79]: This might have came when I was resolving merge conflicts

File: wagtail/admin/templates/wagtailadmin/login.html (hunk 17 above)
Comment (PaarthAgarwal) [reply wagtail/admin/templates/wagtailadmin/login.html:55]: Cool cool

File: wagtail/admin/templates/wagtailadmin/login.html [hunk 19]
```diff
@@ -31,29 +31,24 @@ <h1>{% block branding_login %}{% trans "Sign in to Wagtail" %}{% endblock %}</h1
...
+                            <a class="reset-password" href="{% url 'wagtailadmin_password_reset' %}" aria-label="{% trans 'Reset your password' %}">{% trans "Forgot your password?" %}</a>
+                        {% endif %}
                     </div>
-
-                    {% trans "Remember me" as checkbox_text %}
-                    {% include "wagtailadmin/shared/forms/single_checkbox.html" with name="remember" label_classname="w-label=3 w-mb-4" text=checkbox_text %}
                 {% endblock %}
-
-                <div class="w-mt-4 w-mb-4">
-                    {% block submit_buttons %}
```
[the remark this answers is not in the document]
Comment (PaarthAgarwal) [reply wagtail/admin/templates/wagtailadmin/login.html:49]: The reason for this change was:
We don't need those styles and Dan suggested we shouldn't keep divs if they don't serve any particular purpose or I'd they do we should document them

**Comment by lb-** [comment]
Thanks @PaarthAgarwal  - take a look locally and see if there is anything missing or any visual bugs and we can fix up asap.

**Comment by lb-** [comment]
@PaarthAgarwal small fix as discussed to ensure the sizes fit better & content is not as tall.
573da5d307293605aba0385f5035d986ca8487c1

Also fixed issue where focus outline on password reset link would cut into password field

Take a look and then if there is anything else major we can raise a new issue or a new PR.
